public/audiograph.js

```diff
@@ -181,6 +183,46 @@ class ADSRNode extends AudioNode
     }
 }
 
+/**
+ * BitCrush distortion
+ * See https://github.com/maximecb/noisecraft/issues/81
+ */
+export class BitCrush extends AudioNode
+{
+    constructor(id, state, sampleRate, send)
+    {
+        super(id, state, sampleRate, send);
+        this.reconfigure(state.params.bitdepth);
+    }
+
+    reconfigure(numBits) {
+        console.log(`BitCrush node: ${this.nodeId}, numBits: ${numBits}`);
+        assert(Number.isInteger(numBits) && numBits > 0 && numBits <= 32, `Invalid numBits: ${numBits}`);
+
+        this.nbits = numBits;
+        this.maxInt = 2 ** this.nbits;
+        this.halfMaxInt = this.maxInt / 2;
+        this.invHalfMaxInt = 1 / this.halfMaxInt;
+        this.quantizeDelta = 1 / this.maxInt;
```

Comment:
I don't think it's super useful to store all these constants. Cache bandwidth and the cost of JS property accesses makes it not worth it IMO.

Reply:
My motivation was to (1) reduce the overall number of math operations done in the update cycle, and (2) avoid using the divide operation in update. IIRC this was important at the CPU level and it still seems relevant in js code (ref: https://stackoverflow.com/questions/10535605/javascript-performance-divide-or-multiply-vs).

Computing const values in tight loops makes me feel icky - do you have any refs on why property access is expensive? I tried googling but didn't find anything compelling.